packages/babel-compiler/babel-compiler.js

```diff
@@ -300,6 +300,16 @@ BCp._inferHelper = function (
   merge(babelOptions, babelrc, "presets");
   merge(babelOptions, babelrc, "plugins");
 
+  const babelEnv = (process.env.BABEL_ENV ||
+                    process.env_NODE_ENV ||
```

Comment:
@hexsprite Ben's referring to the typo in your `process.env_NODE_ENV` reference (note the `_`).